### PR TITLE
GH-957 Adds c3chart/Combo.js

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-957

@mlzummo @dtsnell4 Please review

I added a more complex sample to show capabilities of the various combinations of chart types (We should probably do the same with the other libraries:
http://rawgit.com/jbrundage/Visualization/C3-ComboChart/demos/dermatology.html?src/c3chart/Combo

EDIT: Replaced sample URL with one that works

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

File: demos/dermatology.html [hunk 1]
```diff
@@ -296,7 +296,9 @@
                     currWidget = widget;
                     if (params) {
                         for (var key in params) {
-                            currWidget[key](params[key]);
+                            if(typeof(currWidget[key]) === "function"){
```
Comment (mlzummo) [inline demos/dermatology.html]: i suspect there are other areas of of code that can use this same fix in the library?

Comment (jbrundage) [inline demos/dermatology.html]: Potentially

File: demos/dermatology.html [hunk 2]
```diff
@@ -309,6 +311,7 @@
                             displaySerialization(currWidget);
                             displaySerializationText(currWidget);
                             displayThemeText(currWidget);
+                            frame.doResize();
```
Comment (GordonSmith) [inline demos/dermatology.html]: Why is this needed?

File: src/c3chart/Combo.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["./CommonND"], factory);
+    } else {
+        root.c3chart_Column = factory(root.c3chart_CommonND);
```
Comment (GordonSmith) [inline src/c3chart/Combo.js]: Wrong root name

**Comment by GordonSmith** [comment]
Lint is failing

File: demos/dermatology.html (hunk 2 above)
Comment (jbrundage) [inline demos/dermatology.html]: I'm not sure how that ended up there... I'll investigate later today

Comment (jbrundage) [inline demos/dermatology.html]: At one point this was needed otherwise the labels were cut off until after the first update. This issue doesn't seem to be happening anymore. I'll remove this.

**Comment by GordonSmith** [comment]
@jbrundage Sample page is failing to load in Chrome.
@mlzummo @dtsnell4 Please Review

**Comment by jbrundage** [comment]
@GordonSmith I've added a working link to the top comment in this thread

File: src/c3chart/Combo.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        CommonND.call(this);
+
+        this._type = "bar";
+        this._previousTypes = undefined;
+    }
+    Combo.prototype = Object.create(CommonND.prototype);
+    Combo.prototype.constructor = Combo;
+    Combo.prototype._class += " c3chart_Column";
+
+    Combo.prototype.publish("isStacked", false, "boolean", "Stack Chart",null,{tags:["Basic"]});
```
Comment (GordonSmith) [inline src/c3chart/Combo.js]: I don't think "isStacked" does what I expect it to do.  Also in chart/Column we have a property called "stacked" is it the same thing?

File: src/c3chart/Combo.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    Combo.prototype.publish("baseType", "bar", "set", "Base chart type",["bar","line","spline","area","area-spline","step","area-step","scatter"],{tags:["Basic"]});
```
Comment (GordonSmith) [inline src/c3chart/Combo.js]: What is baseType?  looks like the default type?

Comment (jbrundage) [inline src/c3chart/Combo.js]: Yes, it's the default chart type (I'll change it to defaultType, that makes more sense)

File: src/c3chart/Combo.js (hunk 4 above)
Comment (jbrundage) [inline src/c3chart/Combo.js]: Upon further inspection... this Combo chart's "isStacked" mode stacks the values of each individual chart type in the same way that AMCharts stacks their values while in stacking mode.

Comment (jbrundage) [inline src/c3chart/Combo.js]: @GordonSmith 

Notice that the **bar** values are stacked additively while the **line** values aren't effected by stack (because only one data set is using **line**):
http://rawgit.com/jbrundage/Visualization/C3-ComboChart/demos/dermatology.html?src/c3chart/Combo&types=bar,bar,line&isStacked=1

Comment (GordonSmith) [inline src/c3chart/Combo.js]: I suspect my use of "stacked" in hpcc charts is the odd one out.  Can you rename isStacked to stacked?
